include/common/vkwrap/utils.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include "common/vulkan_include.h"
+
+#include "utils/algorithm.h"
+
+#include <range/v3/algorithm/any_of.hpp>
+#include <range/v3/range/concepts.hpp>
+
+namespace vkwrap
+{
+
+template <ranges::range Range>
+inline void
+writeSuitableSharingInfo( auto& create_info, Range&& indices )
+{
+    auto m_unique_indices = utils::getUniqueElements( indices );
+
+    if ( m_unique_indices.size() >= 2 )
+    {
+        create_info.setSharingMode( vk::SharingMode::eConcurrent );
+        create_info.setQueueFamilyIndexCount( static_cast<uint32_t>( m_unique_indices.size() ) );
+        create_info.setPQueueFamilyIndices( m_unique_indices.data() );
+    } else
+    {
+        create_info.setSharingMode( vk::SharingMode::eExclusive );
+        create_info.setQueueFamilyIndexCount( 0 );
+        create_info.setPQueueFamilyIndices( nullptr );
+    }
+} // writeSuitableSharingInfo
```

Comment:
I think this function could accept a range of vkwrap::Queue and extract family indices from those. This would make a better interface

Reply:
It adds some code dependency, but it makes better usage. Ok.